camera/ios/Plugin/CameraPlugin.swift

```diff
@@ -0,0 +1,439 @@
+            call?.resolve([
+                "dataUrl": "data:image/jpeg;base64," + jpeg.base64EncodedString(),
+                "exif": processedImage.exifData,
+                "format": "jpeg"
+            ])
+        } else if settings.resultType == CameraResultType.uri {
+            guard let fileURL = try? saveTemporaryImage(jpeg),
+                  let webURL = bridge?.portablePath(fromLocalURL: fileURL) else {
+                call?.reject("Unable to get portable path to file")
+                return
+            }
+            call?.resolve([
+                "path": fileURL.path,
+                "exif": processedImage.exifData,
+                "webPath": webURL.path,
+                "format": "jpeg"
+            ])
+        }
+    }
+
+    func showPrompt() {
+        // Build the action sheet
+        let alert = UIAlertController(title: settings.userPromptText.title, message: nil, preferredStyle: UIAlertController.Style.actionSheet)
+        alert.addAction(UIAlertAction(title: settings.userPromptText.photoAction, style: .default, handler: { [weak self] (_: UIAlertAction) in
+            self?.showPhotos()
+        }))
+
+        alert.addAction(UIAlertAction(title: settings.userPromptText.cameraAction, style: .default, handler: { [weak self] (_: UIAlertAction) in
+            self?.showCamera()
+        }))
+
+        alert.addAction(UIAlertAction(title: settings.userPromptText.cancelAction, style: .cancel, handler: { [weak self] (_: UIAlertAction) in
+            self?.call?.reject("User cancelled photos app")
+        }))
+
+        self.bridge?.viewController?.present(alert, animated: true, completion: nil)
+    }
+
+    func showCamera() {
+        // check if we have a camera
+        if (bridge?.isSimEnvironment ?? false) || !UIImagePickerController.isSourceTypeAvailable(UIImagePickerController.SourceType.camera) {
+            CAPLog.print("⚡️ ", self.pluginId, "-", "Camera not available in simulator")
+            bridge?.alert("Camera Error", "Camera not available in Simulator")
+            call?.reject("Camera not available while running in Simulator")
+            return
+        }
+        // check for permission
+        let authStatus = AVCaptureDevice.authorizationStatus(for: .video)
+        if authStatus == .restricted || authStatus == .denied {
+            call?.reject("User denied access to camera")
+            return
+        }
+        // we either already have permission or can prompt
+        AVCaptureDevice.requestAccess(for: .video) { [weak self] granted in
+            if granted {
+                DispatchQueue.main.async {
+                    self?.presentCameraPicker()
+                }
+            } else {
+                self?.call?.reject("User denied access to camera")
+            }
+        }
+    }
+
+    func showPhotos() {
+        // check for permission
+        let authStatus = PHPhotoLibrary.authorizationStatus()
+        if authStatus == .restricted || authStatus == .denied {
```

Comment:
Is this the behavior we want? Technically, the picker (either PHPickerViewController on iOS 14 or UIImagePickerController) does not require permission. The picker is presented out-of-process and requires user interaction so even if the user denies library access the application can still show the picker and get an image. The permission allows us to access the underlying asset and get the additional metadata about the image that the user selected. With denied permissions we can still get a picture but it won't have any metadata to return with it. That's potentially confusing and is inconsistent with other platforms and how the camera works, though.